src/test/kotlin/FilePathGeneratorTest.kt

```diff
@@ -0,0 +1,42 @@
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import org.junit.Test
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.assertThrows
+import org.wycliffeassociates.sourceaudio.common.directory.upload.FilePathGenerator
+import java.io.File
+import java.lang.IllegalArgumentException
+
+class FilePathGeneratorTest {
+
+    @Test
+    fun testPathCreationWithValidInput() {
+        val testModels: List<FilePathTestModel> = getTestModelsFromFile("./src/test/resources/TestCases.json")
+
+        for (model in testModels) {
+            assertEquals(
+                model.expectedResult,
+                FilePathGenerator.createPathFromFile(model.getFileUploadModel()),
+                "Model at index: ${testModels.indexOf(model)}"
+            )
+        }
+    }
+
+    @Test
+    fun testPathCreationWithExceptions() {
+        val modelList: List<FilePathTestModel> = getTestModelsFromFile("./src/test/resources/TestCasesException.json")
```

Comment:
I would prefer to see all the test cases in the same source file, with a field that defines whether or not this is expected to throw an exception (and perhaps even which exception).